dlc-manager/tests/ordinals_tests.rs

```diff
@@ -0,0 +1,702 @@
+
+#[derive(Debug, serde::Deserialize, PartialEq, Eq)]
+struct InscriptionsOutput {
+    inscription: String,
+    location: String,
+    explorer: String,
+    postage: u64,
+}
+
+#[macro_use]
+#[allow(dead_code)]
+mod test_utils;
+
+const DEFAULT_POSTAGE: u64 = 10000;
+
+fn ord_binary_path() -> PathBuf {
+    let mut dir = current_dir().unwrap();
+    if !dir.to_str().unwrap().ends_with("dlc-manager") {
+        dir.push("dlc-manager");
+    }
+    dir.push("tests");
+    dir.push("ord");
+    dir
+}
+
+fn ord_command_base() -> Command {
+    let ord_bin_path = ord_binary_path();
+    let mut command = Command::new(&ord_bin_path);
+    command.arg("-r");
+    command.arg("--bitcoin-rpc-user");
+    command.arg("testuser");
+    command.arg("--bitcoin-rpc-pass");
+    command.arg("lq6zequb-gYTdF2_ZEUtr8ywTXzLYtknzWU4nV8uVoo=");
+    command.arg("--data-dir");
+    command.arg("./tests/orddata");
+    command
+}
+
+fn ord_command_wallet(name: &str) -> Command {
+    let mut command = ord_command_base();
+    command.arg("--wallet");
+    command.arg(name);
+    command.arg("wallet");
+    command
+}
+
+fn create_ordinal_wallet(name: &str) {
+    let mut command = ord_command_wallet(name);
+    command.arg("create");
+    let mut handle = command.spawn().expect("not to fail");
+    handle.wait().unwrap();
+}
+
+fn inscribe_logo(wallet_name: &str) {
+    let mut command = ord_command_wallet(wallet_name);
+    command.arg("inscribe");
+    command.arg("--fee-rate");
+    command.arg("1");
+    command.arg("--file");
+    command.arg("/ord/logo-bw.svg");
+    let mut handle = command.spawn().expect("not to fail");
+    handle.wait().unwrap();
+}
+
+fn get_inscriptions(wallet_name: &str) -> Vec<InscriptionsOutput> {
+    let mut command = ord_command_wallet(wallet_name);
+    command.arg("inscriptions");
+    let res = command.output().expect("not to fail");
+    serde_json::from_str(&String::from_utf8(res.stdout).unwrap()).unwrap()
+}
+
+fn generate_blocks(sink_rpc: &Client, nb_blocks: u64) {
+    let prev_blockchain_height = sink_rpc.get_block_count().unwrap();
+
+    let sink_address = sink_rpc
+        .call("getnewaddress", &["".into(), "bech32m".into()])
+        .expect("RPC Error");
+    sink_rpc
+        .generate_to_address(nb_blocks, &sink_address)
+        .expect("RPC Error");
+
+    // Make sure all blocks have been generated
+    let mut cur_blockchain_height = prev_blockchain_height;
+    while cur_blockchain_height < prev_blockchain_height + nb_blocks {
+        std::thread::sleep(std::time::Duration::from_millis(200));
+        cur_blockchain_height = sink_rpc.get_block_count().unwrap();
+    }
+}
+
+fn get_enum_ord_outcome_descriptor(win_both: bool) -> OrdOutcomeDescriptor {
+    let outcome_payouts: Vec<_> = enum_outcomes()
+        .iter()
+        .enumerate()
+        .map(|(i, x)| {
+            let payout = if i % 2 == 0 {
+                Payout {
+                    offer: TOTAL_COLLATERAL,
+                    accept: 0,
```

Comment:
This logic feels a bit fragile, here and down where creating the OrdEnumDescriptor to_offer_payouts. the i % 2 logic in both places takes some brainpower to follow and a bug could sneak it's way in.  maybe there's some way to hard-code the outcomes in a clearer way.

Reply:
Yeah there are many ways to decide the payouts depending on the outcome. This one is just giving everything to one party for outcomes in odd position and the other party for outcomes in even positions. I thought it nice because it's not too simple while still not being too complicated IMHO (if we do e.g. the first half to one party and the other half to the other party it feels easier to miss some possible issues). But I don't have a strong opinion so let me know what you would prefer.

Reply:
OK, I'll review it more and think about whether I could like another design better. Maybe I'll propose something in a PR and we can discuss. Thanks for the feedback. Anyway, this isn't a blocker